backend/__tests__/queries.test.js

```diff
@@ -0,0 +1,323 @@
+import request from "supertest";
+import app from "../app";
+
+describe("Test GraphQL queries", () => {
+	it("retrieves the proper archipelagos full response", async () => {
+		const query = `{ 
+      archipelagos{
+        id
+        name 
+        islands{
+          id
+          name
+          nativeFruit
+          islanders{
+            id
+            name
+          }
+        }
+      } 
+    }`;
+		const expectedResponse = {
+			data: {
+				archipelagos: [
+					{
+						id: "1",
+						islands: [
+							{
+								id: "1",
+								islanders: [
+									{ id: "1", name: "Francis" },
+									{ id: "3", name: "Corrina" },
+								],
+								name: "Raftel",
+								nativeFruit: "Peaches",
+							},
+							{
+								id: "2",
+								islanders: [{ id: "2", name: "Riki" }],
+								name: "Montoya",
+								nativeFruit: "Apples",
+							},
+						],
+						name: "Covidian Montréal Consortium",
+					},
+					{
+						id: "2",
+						islands: [
+							{
+								id: "3",
+								islanders: [{ id: "4", name: "Bawb" }],
+								name: "Syracuse",
+								nativeFruit: "Oranges",
+							},
+						],
+						name: "Secret Group of Destiny",
+					},
+				],
+			},
+		};
+
+		const res = await request(app).post("/graphql").send({ query: query });
+		const parsedRes = JSON.parse(res.text);
+
+		expect(res.statusCode).toEqual(200);
+		expect(parsedRes).toEqual(expectedResponse);
+	});
+
+	it("retrieves the proper archipelago full response by archipelagoId", async () => {
+		const query = `{ 
+      archipelago(archipelagoId: 1){
+        id
+        name 
+        islands{
+          id
+          name
+          nativeFruit
+          islanders{
+            id
+            name
+          }
+        }
+      } 
+    }`;
+		const expectedResponse = {

```

Comment:
Maybe we could move this variable definition to a `before()` or to an outer scope? It's reused a few times throughout and would help keep the code DRY

Reply:
The name expectedResponse is indeed used in each test, however the value is different. Wouldn't make sense